syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
+                        'config')
+                else:
+                    remote_output_conf = []
+                if len(gateway.get('shutters', [])) > 0:
+                    remote_shutter_conf = self._call_remote('get_shutter_configurations', self._gateways[ip]).get(
+                        'config')
+                else:
+                    remote_shutter_conf = []
             except Exception as ex:
-                logger.exception('Could not load temperature mapping: {0}'.format(ex))
+                raise RuntimeError(f"Couldn't connect to GW {ip}: {ex}")
 
-        self._output_mapping = {}
-        for entry in self._config.get('outputs', []):
+            # Specific for each remote GW as this is remote 2 local. A loop over every remote GW will occur and this
+            # will give our local sensor_dto if a value has changed.
+            sensor_mapping = {}
+            for entry in gateway.get('sensors', []):
+                try:
+                    remote_sensor_id = entry.get('remote_sensor_id', -1)
+                    if remote_sensor_id < 0:
+                        raise RuntimeError("Please give valid sensor ids")
+                    external_id = f"syncer/{ip}/{remote_sensor_id}"
+                    for remote_sensor in remote_sensor_conf:
+                        if remote_sensor.get('id') == remote_sensor_id:
+                            sensor = self.connector.sensor.register(external_id=external_id,
+                                                                    physical_quantity=remote_sensor.get(
+                                                                        'physical_quantity'),
+                                                                    unit=remote_sensor.get('unit'),
+                                                                    name=f"{gw_name}/{remote_sensor.get('name')}")
+                            sensor_mapping[remote_sensor_id] = sensor
+                except Exception as ex:
+                    logger.exception(f'Could not load sensor mapping for GW with ip {ip}: {ex}')
+
+            self._gateways[ip]["sensor_mapping"] = sensor_mapping
+
+            # Global output_mapping variable because this is local 2 remote, thus if local output changes -> mapping to
+            # remote outputs will happen. One specific local output changes to different remote output changes
+            local_output_conf = json.loads(self.webinterface.get_output_configurations()).get('config')
+
+            for entry in gateway.get('outputs', []):
+                try:
+                    # Organise local output data and their mapping with one or more remote outputs by making use of a dict
+                    local_id = int(entry.get('local_output_id', -1))
+                    remote_id = int(entry.get('remote_output_id', -1))
+                    if local_id < 0 or remote_id < 0:
+                        raise RuntimeError("Please give valid output ids")
```

Comment:
maybe just log an error instead of raising an exception. When raising an exception, the plugin will not run, but the rest of the correct config will also not run. When just logging a verbose error which local/remote id is incorrect, you can just skip that one and cary on.

Reply:
This runtime error will be catched later on and will be logged as info. The plugin will keep on going in that manner

Reply:
This is done for a more complete error handling. If another error occurs, that one will get caught, but this error is not known to the normal python error class, thus we'll raise it ourselves